src/GeoJson.php

```diff
@@ -160,7 +170,7 @@ final public static function jsonUnserialize($json): self
             $args[] = CoordinateReferenceSystem::jsonUnserialize($json['crs']);
         }
 
-        $class = sprintf('GeoJson\%s\%s', (strncmp('Feature', $type, 7) === 0 ? 'Feature' : 'Geometry'), $type);
+        $class = sprintf('GeoJson\%s\%s', (strncmp(self::TYPE_FEATURE, $type, 7) === 0 ? self::TYPE_FEATURE : 'Geometry'), $type);
```

Comment:
```suggestion
        $class = sprintf('GeoJson\%s\%s', (strncmp('Feature', $type, 7) === 0 ? 'Feature' : 'Geometry'), $type);
```

I'd prefer to revert this since "Feature" is not referring to the actual type in this context. We're just doing a prefix check on the type string.

Alternatively, we could calculate the namespace component in advance (e.g. during the `switch` or a separate condition) but that just adds more complexity for the same result.